.gitlab/docker/Dockerfile.build

```
@@ -0,0 +1,29 @@
+ARG BASE_IMAGE=docker.io/pikaorg/pika-ci-base:17
+ARG SOURCE_DIR=/pika/source
+ARG BUILD_DIR=/pika/build
+
+FROM $BASE_IMAGE
+
+ARG SOURCE_DIR
+ARG BUILD_DIR
```

Maybe?
```suggestion
ARG BASE_IMAGE
FROM $BASE_IMAGE

ARG SOURCE_DIR=/pika/source
ARG BUILD_DIR=/pika/build
```
At least we don't need `SOURCE_DIR` and `BUILD_DIR` twice as `ARG`s. Do you think it makes sense to move the definitions of `SOURCE_DIR` and `BUILD_DIR` to `pipeline.yml` and not have any default values here? Not a blocker.